homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,486 @@
+            return str(returnvalue).capitalize()
+        elif self.device_type == "HotWater_Mode":
+            currentmode = self.session.hotwater.get_mode(self.node_id)
+            if currentmode == "SCHEDULE":
+                return str(STATE_AUTO).capitalize()
+            elif currentmode == "ON":
+                return str(STATE_ON).capitalize()
+            elif currentmode == "OFF":
+                return str(STATE_OFF).capitalize()
+        elif self.device_type == "HotWater_Boost":
+            returnvalue = self.session.hotwater.get_boost(self.node_id)
+            return str(returnvalue).capitalize()
+        elif self.device_type == "Hive_Device_BatteryLevel":
+            self.batt_lvl = self.session.sensor.battery_level(self.node_id)
+            return self.batt_lvl
+        elif self.device_type == "Hive_Device_Sensor":
+            return self.session.sensor.get_state(self.node_id,
+                                                 self.node_device_type)
+        elif self.device_type == "Hive_Device_Light_Mode":
+            return self.session.sensor.get_mode(self.node_id)
+        elif self.device_type == "Hive_Device_Plug_Mode":
+            return self.session.sensor.get_mode(self.node_id)
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        if self.device_type == "Heating_CurrentTemperature":
+            return self.get_current_temp_sa()
+        elif self.device_type == "Heating_TargetTemperature":
+            return None
+        elif self.device_type == "Heating_State":
+            return self.get_heating_state_sa()
+        elif self.device_type == "Heating_Mode":
+            return self.get_heating_state_sa()
+        elif self.device_type == "Heating_Boost":
+            s_a = {}
+            if self.session.heating.get_boost(self.node_id) == "ON":
+                minsend = self.session.heating.get_boost_time(self.node_id)
+                s_a.update({"Boost ends in":
+                            (str(minsend) + " minutes")})
+            return s_a
+        elif self.device_type == "HotWater_State":
+            return self.get_hotwater_state_sa()
+        elif self.device_type == "HotWater_Mode":
+            return self.get_hotwater_state_sa()
+        elif self.device_type == "HotWater_Boost":
+            s_a = {}
+            if self.session.hotwater.get_boost(self.node_id) == "ON":
+                endsin = self.session.hotwater.get_boost_time(self.node_id)
+                s_a.update({"Boost ends in":
+                            (str(endsin) + " minutes")})
+            return s_a
+        elif self.device_type == "Hive_Device_BatteryLevel":
+            return None
+        elif self.device_type == "Hive_Device_Sensor":
+            return None
+        elif self.device_type == "Hive_Device_Light_Mode":
+            return None
+        elif self.device_type == "Hive_Device_Plug_Mode":
+            return None
+
+    def get_current_temp_sa(self):

```

Comment:
This should be extracted to pyhiveapi library.

Reply:
why? this is an HA specific thing

Reply:
done